.github/workflows/build-test.yml

```diff
@@ -0,0 +1,110 @@
+  build-tests:
+    name:                       Test and Build
+    strategy:
+      matrix:
+        platform:
+          - ubuntu-16.04
+          - macos-latest
+          - windows-latest
+        toolchain:
+          - stable
+    runs-on:                    ${{ matrix.platform }}
+    steps:
+      - name:                   Checkout sources
+        uses:                   actions/checkout@v2
+        with:
+          fetch-depth:          50
+      - name:                   Checkout submodules
+        shell:                  bash
+        run:                    git submodule update --init --recursive
+      - name:                   Install toolchain
+        uses:                   actions-rs/toolchain@v1
+        with:
+          toolchain:            ${{ matrix.toolchain }}
+          override:             true
+      - name:                   Cache cargo registry
+        uses:                   actions/cache@v1.1.2
+        with:
+          path:                 ~/.cargo/registry
+          key:                  ${{ runner.os }}-cargo-registry-build-tests-${{ hashFiles('**/Cargo.lock') }}
+      - name:                   Cache cargo index
+        uses:                   actions/cache@v1.1.2
+        with:
+          path:                 ~/.cargo/git
+          key:                  ${{ runner.os }}-cargo-git-build-tests-${{ hashFiles('**/Cargo.lock') }}
+      - name:                   Cache cargo build
+        uses:                   actions/cache@v1.1.2
+        with:
+          path:                 target
+          key:                  ${{ runner.os }}-cargo-build-target-build-tests-${{ hashFiles('**/Cargo.lock') }}
+      - name:                   Cache sccache linux
+        if:                     matrix.platform == 'ubuntu-16.04'
+        uses:                   actions/cache@v1.1.2
+        with:
+          path:                 "/home/runner/.cache/sccache"
+          key:                  ${{ runner.os }}-sccache-build-tests-${{ hashFiles('**/Cargo.lock') }}
+      - name:                   Cache sccache MacOS
+        if:                     matrix.platform == 'macos-latest'
+        uses:                   actions/cache@v1.1.2
+        with:
+          path:                 "/Users/runner/Library/Caches/Mozilla.sccache"
+          key:                  ${{ runner.os }}-sccache-build-tests-${{ hashFiles('**/Cargo.lock') }}
+      - name:                   Cache sccache Windows
+        if:                     matrix.platform == 'windows-latest'
+        uses:                   actions/cache@v1.1.2
+        with:
+          path:                 "C:\\Users\\runneradmin\\AppData\\Local\\Mozilla\\sccache\\cache"
+          key:                  ${{ runner.os }}-sccache-build-tests-${{ hashFiles('**/Cargo.lock') }}
+      - name:                   Install sccache for ${{ matrix.platform }}
+        env:
+          SCCACHE_CACHE_SIZE:   "1G"
+          SCCACHE_IDLE_TIMEOUT: 0
+        shell:                  pwsh
```

Comment:
in my opinion would be cleaner to keep these scripts as separate files so the main CI file is easier to read

Reply:
Previously, we supported [Docker scripts](https://github.com/paritytech/scripts/tree/master/dockerfiles) for building our binaries (parity-ci-*). But now it is not known who will support them, so at this stage I made CI without using Docker images. And we haven't used Docker caching before.
Caching is now set up quite well.
The time spent on different stages is displayed in the build [stages](https://github.com/OpenEthereum/open-ethereum/pull/11528/checks?check_run_id=477196383) - right column in the log. And total [time](https://github.com/OpenEthereum/open-ethereum/actions/runs/47385478)